Dockerfile

```diff
@@ -1,15 +1,45 @@
-FROM rust:1.64 as builder
+# Define ARG for build platform
+FROM --platform=$BUILDPLATFORM rust:1.64 as builder
+
+# Set ARG for target platform
+ARG TARGETPLATFORM
 
 # Set working directory
 WORKDIR /usr/src/rpc
 
 # Copy source code
 COPY . .
 
-RUN rustup self update
-
-# Build the application
-RUN cargo build --all --release
+# Cross-compile the application for a given platform
+RUN build_platform() { \
+        ARCH=$1; \
+        COMPILER=$2; \
+        LINKER=$3; \
+        echo "Building for $TARGETPLATFORM"; \
+         # Add the specified Rust target architecture
+        rustup target add $ARCH; \
+        # Update package lists and install the specified compiler
+        apt-get update && apt-get -y install $COMPILER; \
+        # Build the Rust application for the specified target
+        cargo build --all --release \
+          --target=$ARCH \
+          --config target.$ARCH.linker=\"$LINKER\"; \
```

Comment:
Here we cross-compile to the target architecture, thanks @bajpai244 for the previous work with [katana](https://github.com/kkrt-labs/kakarot/pull/594/files)